src/System.Diagnostics.Debug/src/System/Diagnostics/Debug.Unix.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.InteropServices;
+
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Diagnostics
+{
+    public static partial class Debug
+    {
+        // Internal so that the tests can override this.
+        internal static IDebugLogger s_logger = new UnixDebugLogger();
+
+        // --------------
+        // PAL ENDS HERE
+        // --------------
+
+        internal class UnixDebugLogger : IDebugLogger
+        {
+            public void ShowAssertDialog(string stackTrace, string message, string detailMessage)
+            {
+                // TODO: Implement this
+                throw new NotImplementedException();
+            }
+
+            public void WriteLineCore(string message)
+            {
+                message = message + "\n";
+                Write(message);
+            }
+
+            public void WriteCore(string message)
+            {
+                // We don't want to write UTF-16 to standard error.  Ideally we would transcode this
+                // to UTF8, but the downside of that is it pulls in a bunch of stuff into what is ideally
+                // a path with minimal dependencies (as to prevent re-entrency), so we'll take the strategy
+                // of just throwing away any non ASCII characters from the message and writing the rest
+
+                const int BufferLength = 256;
+
+                unsafe
+                {
+                    byte* buf = stackalloc byte[BufferLength];
+                    int bufCount = 0;
+                    int i = 0;
+
+                    using (SafeFileHandle stderr = SafeFileHandle.Open("/dev/stderr", Interop.libc.OpenFlags.O_WRONLY, 0))
```

Comment:
Since we now use this path in a few places, we could consider adding it (and the others like it) as constants in some Common file somewhere, or potentially even add helper methods like OpenStderr to the Common partial definition of SafeFileHandle.